### DISCUSSION TITLE
How to lock version of preinstalled software in GitHub Actions Virtual Environments

### DISCUSSION THREAD

**Discussion by mohammadsiyou** [community/community#26740] [Actions]
<p>Hi,<br>
As we know, preinstalled software in GitHub actions virtual environment are updated weekly.</p>
<p>So I want to lock version of some specific software in order to not be updated.</p>
<p>For example I use Chrome for running visual regression test and the images that taken by Chrome 95 is really different than those ones taken by Chrome 96.</p>
<p>Moreover, Maybe on new version of Chrome my tests would go haywire!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="mohammadsiyou" data-post="1" data-topic="215387">
<div class="title">
<div class="quote-controls"></div>
 mohammadsiyou:</div>
<blockquote>
<p>So I want to lock version of some specific software in order to not be updated.</p>
</blockquote>
</aside>
<p>You can’t. If you want to define your own environment, you need to use a self-hosted runner.</p>
<aside class="quote no-group" data-username="mohammadsiyou" data-post="1" data-topic="215387">
<div class="title">
<div class="quote-controls"></div>
 mohammadsiyou:</div>
<blockquote>
<p>Moreover, Maybe on new version of Chrome my tests would go haywire!</p>
</blockquote>
</aside>
<p>If your site breaks in an updated browser that means it’ll break for most users, too, so you’ll have to update quickly either way. So I’d say it’s a good thing if the tests help you notice the problem as soon as possible. :wink:</p>
